nrel/hive/model/vehicle/mechatronics/powertrain/tabular_powertrain.py

```diff
@@ -43,16 +42,9 @@ def from_data(
             if key not in data:
                 raise IOError(f"invalid input file for tabular power train model missing key {key}")
 
-        if not valid_unit(data["speed_units"]):
-            raise TypeError(f"{data['speed_units']} not a recognized unit in hive")
-        elif not valid_unit(data["distance_units"]):
-            raise TypeError(f"{data['distance_units']} not a recognized unit in hive")
-        elif not valid_unit(data["energy_units"]):
-            raise TypeError(f"{data['energy_units']} not a recognized unit in hive")
-
-        speed_units = data["speed_units"]
-        energy_units = data["energy_units"]
-        distance_units = data["distance_units"]
+        speed_units = Unit.from_string(data["speed_units"])
```

Comment:
try/catch and chain so we can tell user "failure reading tabular powertrain input" on Unit.from_string exception.

Reply:
yes, that's a great idea, added in